demos/gaze_estimation_demo/cpp_gapi/main.cpp

```diff
@@ -0,0 +1,342 @@
+        /** Detect states of right eyes **/
+        cv::GArray<cv::GMat> right_state_eyes =
+            cv::gapi::infer2<nets::Eyes>(in, right_processed_eyes);
+        /** Recognize states of eyes **/
+        cv::GArray<int> state_left_eyes, state_right_eyes;
+        std::tie(state_left_eyes,  // open/closed
+                 state_right_eyes) // 1   /0
+            = custom::ProcessEyes::on(in, left_state_eyes, right_state_eyes);
+
+        /** Prepare eyes for gaze-estimation network **/
+        std::tie(left_processed_eyes,
+                 right_processed_eyes) = custom::PrepareEyes::on(in,
+                                                                 left_eyes_rc,
+                                                                 right_eyes_rc,
+                                                                 angles_r,
+                                                                 cv::Size{60, 60});
+
+        /** Gaze estimation **/
+        cv::GArray<cv::GMat> gaze_vectors = cv::gapi::infer2<nets::Gaze>(in,
+                                                                         left_processed_eyes,
+                                                                         right_processed_eyes,
+                                                                         heads_pos_without_roll);
+        /** Processing gaze estimation results **/
+        cv::GArray<cv::Point3f> processed_gaze_vectors =
+            custom::ProcessGazes::on(gaze_vectors, angles_r);
+
+        /** Inputs and outputs of graph **/
+        cv::GComputation graph(cv::GIn(in),
+                               cv::GOut(cv::gapi::copy(in),
+                                        faces_conf,
+                                        faces_rc,
+                                        faces_landmarks,
+                                        heads_pos,
+                                        left_eyes_rc,
+                                        right_eyes_rc,
+                                        leftEyeMidpoint,
+                                        rightEyeMidpoint,
+                                        state_left_eyes,
+                                        state_right_eyes,
+                                        processed_gaze_vectors));
+        /** ---------------- End of graph ---------------- **/
+        /** Configure networks **/
+        auto face_net = cv::gapi::ie::Params<nets::Faces> {
+            FLAGS_m_fd,                          // path to topology IR
+            fileNameNoExt(FLAGS_m_fd) + ".bin",  // path to weights
+            FLAGS_d_fd,                          // device specifier
+        };
+        /** Get information about frame from cv::VideoCapture **/
+        std::shared_ptr<ImagesCapture> cap = openImagesCapture(FLAGS_i, FLAGS_loop, 0,
+            std::numeric_limits<size_t>::max(), stringToSize(FLAGS_res));
+        const auto tmp = cap->read();
```

Comment:
```suggestion
        const auto tmp = cap->read();
        cap.reset();
```
It can't reopen `0` otherwise.

Reply:
Added, thanks.